internal/updater/updater_test.go

```diff
@@ -0,0 +1,457 @@
+package updater
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
```

Comment:
Do we want to use the `fake` client instead of `envtest`? The main reason I ask is this line from [the package page for it](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/fake#:~:text=When%20in%20doubt%2C%20it%27s%20almost%20always%20better%20not%20to%20use%20this%20package%20and%20instead%20use%20envtest.Environment%20with%20a%20real%20client%20and%20API%20server.):

    When in doubt, it's almost always better not to use this package and instead use envtest.Environment with a real client and API server.

Reply:
Fake is enough for here but I'll look at envtest.   Thanks!

Reply:
For now, I think the fake is enough.   I'll look at envtest again once the storage unit test PR is merged.  The 2 different clients may confuse unit test development later.

Reply:
I think that the `fake` client makes sense here specifically since a lot of these tests don't rely on core kube API's. The storage unit test is constructing/retrieving ConfigMaps under the hood so I figured it would be easier to just use envtest there. I agree that may get confusing moving forward though.